### DISCUSSION TITLE
[mini.diff] signs are not returned in sign_getplaced?

### DISCUSSION THREAD

**Discussion by ahmedelgabri** [nvim-mini/mini.nvim#789] [Q&A] [answered]
I'm trying to build a custom `statuscolumn` & wanted to show the diff signs, but when I call `sign_getplaced` on a modified line it doesn't return the diff signs. I have explicitly set `view.style = 'sign'`

This is how I'm quickly testing this 

```vim
:echo sign_getplaced(bufname(), {'group': '*', 'lnum': v:lnum})
```

So how can I get signs added by `mini.diff`?

**Comment by echasnovski** [comment, marked as answer]
Signs in 'mini.diff' are placed with extmarks (which are usually more efficient). The supposed way to retrieve them is with `nvim_buf_get_extmarks()`. Since neovim/neovim#25724 (i.e. on current Nightly) old style signs can also be retrieved this way. And this also seems to be a preferred way to do this task by ['statuscolumn' author](https://github.com/luukvbaal/statuscol.nvim/blob/d954893262a57a92e46edd87de67e2b3fe72305e/lua/statuscol.lua#L236).

So the suggested way would be to get all signs and filter later by `sign_hl_group`:

```lua
vim.api.nvim_buf_get_extmarks(0, -1, { vim.v.lnum - 1, 0}, { vim.v.lnum - 1, -1}, { type = 'sign', details = true })
```

The better approach would be to get namespace id of 'mini.diff', but it is not guaranteed to not break. However, for personal usage it might be fine. Currently namespace name [is `MiniDiffViz`](https://github.com/echasnovski/mini.nvim/blob/32b2db5ef9066f3d7ca375ea1ccc0fc44090727d/lua/mini/diff.lua#L825), so signs from 'mini.diff' can be obtained like this:

```lua
-- Evaluate somewhere only once
local diff_ns_id = vim.api.nvim_get_namespaces().MiniDiffViz

-- Get signs only from 'mini.diff'
vim.api.nvim_buf_get_extmarks(0, diff_ns_id, { vim.v.lnum - 1, 0}, { vim.v.lnum - 1, -1}, { type = 'sign', details = true })
```

------

For completeness, there is also [`get_buf_data()`](https://github.com/echasnovski/mini.nvim/blob/32b2db5ef9066f3d7ca375ea1ccc0fc44090727d/doc/mini-diff.txt#L470) which contains `hunks` array, but this might be not the best fit for statusline.

**Reply by ahmedelgabri** [reply]
Thanks @echasnovski
